BENCH_DATA M4_COMP_DEMOGRAPHICS <Bench.TS_datasets.cTimeSeriesDatasetSpec object at 0x7f3e6316ecc0>
TIME :  Date N= 8 H= 6 HEAD= [0 1 2 3 4] TAIL= [3 4 5 6 7]
SIGNAL :  D0102 N= 8 H= 6 HEAD= [ 2.   1.9  1.8  1.8  1.6] TAIL= [ 1.8  1.6  1.5  1.4  1.3]
   D0102  Date
0    2.0     0
1    1.9     1
2    1.8     2
3    1.8     3
4    1.6     4
<class 'pandas.core.frame.DataFrame'>
Int64Index: 8 entries, 0 to 7
Data columns (total 2 columns):
D0102    8 non-null float64
Date     8 non-null int64
dtypes: float64(1), int64(1)
memory usage: 192.0 bytes
None
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 140, in trainModel
    lAutoF1.train(df , self.mTSSpec.mTimeVar , iSignal, iHorizon)
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 25, in train
    self.mSignalDecomposition.train(iInputDS, iTime, iSignal, iHorizon, iExogenousData);
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 459, in train
    logger.info("START_TRAINING '" + iSignal + "'")
Message: "START_TRAINING 'D0102'"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 140, in trainModel
    lAutoF1.train(df , self.mTSSpec.mTimeVar , iSignal, iHorizon)
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 25, in train
    self.mSignalDecomposition.train(iInputDS, iTime, iSignal, iHorizon, iExogenousData);
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 481, in train
    logger.info("END_TRAINING_TIME_IN_SECONDS '" + iSignal + "' " + str(self.mTrainingTime))
Message: "END_TRAINING_TIME_IN_SECONDS 'D0102' 4.981244087219238"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 85, in getInfo
    logger.info("TIME_DETAIL " + self.mTrend.mTimeInfo.info());
Message: "TIME_DETAIL TimeVariable='Date' TimeMin=0 TimeMax=7 TimeDelta=1.0 Estimation = (0 , 8) Validation = (0 , 8) Test = (0 , 8) Horizon=6"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 86, in getInfo
    logger.info("SIGNAL_DETAIL " + self.info());
Message: "SIGNAL_DETAIL SignalVariable='Diff_D0102' Min=-0.2 Max=0.0  Mean=-0.0875 StdDev=0.0599478940414"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 87, in getInfo
    logger.info("BEST_TRANSOFORMATION_TYPE '" + self.mTransformation.get_name("") + "'");
Message: "BEST_TRANSOFORMATION_TYPE 'Diff_'"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 88, in getInfo
    logger.info("BEST_DECOMPOSITION  '" + self.mOutName + "' [" + self.getFormula() + "]");
Message: "BEST_DECOMPOSITION  'Diff_D0102_LinearTrend_residue_zeroCycle_residue_NoAR' [LinearTrend + NoCycle + NoAR]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 89, in getInfo
    logger.info("TREND_DETAIL '" + self.mTrend.mOutName + "' [" + self.mTrend.mFormula + "]");
Message: "TREND_DETAIL 'Diff_D0102_LinearTrend' [LinearTrend]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 90, in getInfo
    logger.info("CYCLE_DETAIL '"+ self.mCycle.mOutName + "' [" + self.mCycle.mFormula + "]");
Message: "CYCLE_DETAIL 'Diff_D0102_LinearTrend_residue_zeroCycle' [NoCycle]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 91, in getInfo
    logger.info("AUTOREG_DETAIL '" + self.mAR.mOutName + "' [" + self.mAR.mFormula + "]");
Message: "AUTOREG_DETAIL 'Diff_D0102_LinearTrend_residue_zeroCycle_residue_NoAR' [NoAR]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 92, in getInfo
    logger.info("MODEL_MAPE MAPE_Fit=" + str(self.mFitPerf.mMAPE) + " MAPE_Forecast=" + str(self.mForecastPerf.mMAPE)  + " MAPE_Test=" + str(self.mTestPerf.mMAPE) );
Message: 'MODEL_MAPE MAPE_Fit=0.0177 MAPE_Forecast=0.0177 MAPE_Test=0.0177'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 93, in getInfo
    logger.info("MODEL_MASE MASE_Fit=" + str(self.mFitPerf.mMASE) + " MASE_Forecast=" + str(self.mForecastPerf.mMASE)  + " MASE_Test=" + str(self.mTestPerf.mMASE) );
Message: 'MODEL_MASE MASE_Fit=0.3266 MASE_Forecast=0.3266 MASE_Test=0.3266'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 94, in getInfo
    logger.info("MODEL_L1 L1_Fit=" + str(self.mFitPerf.mL2) + " L1_Forecast=" + str(self.mForecastPerf.mL2)  + " L1_Test=" + str(self.mTestPerf.mL1) );
Message: 'MODEL_L1 L1_Fit=0.0498366149213 L1_Forecast=0.0498366149213 L1_Test=0.032657967033'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 95, in getInfo
    logger.info("MODEL_L2 L2_Fit=" + str(self.mFitPerf.mL2) + " L2_Forecast=" + str(self.mForecastPerf.mL2)  + " L2_Test=" + str(self.mTestPerf.mL2) );
Message: 'MODEL_L2 L2_Fit=0.0498366149213 L2_Forecast=0.0498366149213 L2_Test=0.0498366149213'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 96, in getInfo
    logger.info("MODEL_COMPLEXITY " + str(self.getComplexity()) );
Message: 'MODEL_COMPLEXITY 48'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 97, in getInfo
    logger.info("AR_MODEL_DETAIL_START");
Message: 'AR_MODEL_DETAIL_START'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 99, in getInfo
    logger.info("AR_MODEL_DETAIL_END");
Message: 'AR_MODEL_DETAIL_END'
Arguments: ()
   Transformation                                              Model  \
0      Diff_D0102  Diff_D0102_LinearTrend_residue_zeroCycle_resid...   
1      Diff_D0102  Diff_D0102_LinearTrend_residue_bestCycle_byL2_...   
2      Diff_D0102  Diff_D0102_PolyTrend_residue_zeroCycle_residue...   
3      Diff_D0102  Diff_D0102_PolyTrend_residue_bestCycle_byL2_re...   
4   RelDiff_D0102  RelDiff_D0102_PolyTrend_residue_zeroCycle_resi...   
5   RelDiff_D0102  RelDiff_D0102_PolyTrend_residue_bestCycle_byL2...   
6      Diff_D0102  Diff_D0102_ConstantTrend_residue_zeroCycle_res...   
7      Diff_D0102  Diff_D0102_ConstantTrend_residue_bestCycle_byL...   
8          _D0102    _D0102_PolyTrend_residue_zeroCycle_residue_NoAR   
9          _D0102  _D0102_PolyTrend_residue_bestCycle_byL2_residu...   
10         _D0102    _D0102_Lag1Trend_residue_zeroCycle_residue_NoAR   
11         _D0102  _D0102_Lag1Trend_residue_bestCycle_byL2_residu...   
12     Diff_D0102  Diff_D0102_Lag1Trend_residue_zeroCycle_residue...   
13  RelDiff_D0102  RelDiff_D0102_Lag1Trend_residue_zeroCycle_resi...   
14     Diff_D0102  Diff_D0102_Lag1Trend_residue_bestCycle_byL2_re...   
15  RelDiff_D0102  RelDiff_D0102_Lag1Trend_residue_bestCycle_byL2...   
16  RelDiff_D0102  RelDiff_D0102_LinearTrend_residue_zeroCycle_re...   
17  RelDiff_D0102  RelDiff_D0102_LinearTrend_residue_bestCycle_by...   
18         _D0102  _D0102_LinearTrend_residue_zeroCycle_residue_NoAR   
19         _D0102  _D0102_LinearTrend_residue_bestCycle_byL2_resi...   
20  RelDiff_D0102  RelDiff_D0102_ConstantTrend_residue_zeroCycle_...   
21  RelDiff_D0102  RelDiff_D0102_ConstantTrend_residue_bestCycle_...   
22         _D0102  _D0102_ConstantTrend_residue_zeroCycle_residue...   
23         _D0102  _D0102_ConstantTrend_residue_bestCycle_byL2_re...   
24   CumSum_D0102  CumSum_D0102_Lag1Trend_residue_zeroCycle_resid...   
25   CumSum_D0102  CumSum_D0102_Lag1Trend_residue_bestCycle_byL2_...   
26   CumSum_D0102  CumSum_D0102_PolyTrend_residue_zeroCycle_resid...   
27   CumSum_D0102  CumSum_D0102_PolyTrend_residue_bestCycle_byL2_...   
28   CumSum_D0102  CumSum_D0102_LinearTrend_residue_zeroCycle_res...   
29   CumSum_D0102  CumSum_D0102_LinearTrend_residue_bestCycle_byL...   
30   CumSum_D0102  CumSum_D0102_ConstantTrend_residue_zeroCycle_r...   
31   CumSum_D0102  CumSum_D0102_ConstantTrend_residue_bestCycle_b...   

    Complexity  FitCount     FitL1     FitL2  FitMAPE  FitMASE  ForecastCount  \
0           48         8  0.032658  0.049837   0.0177   0.3266              8   
1           56         8  0.032658  0.049837   0.0177   0.3266              8   
2           48         8  0.033130  0.046636   0.0183   0.3313              8   
3           56         8  0.033130  0.046636   0.0183   0.3313              8   
4           48         8  0.049051  0.059106   0.0288   0.4905              8   
5           56         8  0.049051  0.059106   0.0288   0.4905              8   
6           32         8  0.056250  0.072349   0.0313   0.5625              8   
7           40         8  0.056250  0.072349   0.0313   0.5625              8   
8           16         8  0.073257  0.080617   0.0444   0.7326              8   
9           24         8  0.073257  0.080617   0.0444   0.7326              8   
10          32         8  0.087500  0.106066   0.0560   0.8750              8   
11          40         8  0.087500  0.106066   0.0560   0.8750              8   
12          64         8  0.087500  0.106066   0.0560   0.8750              8   
13          64         8  0.087500  0.106066   0.0560   0.8750              8   
14          72         8  0.087500  0.106066   0.0560   0.8750              8   
15          72         8  0.087500  0.106066   0.0560   0.8750              8   
16          48         8  0.105128  0.120099   0.0614   1.0513              8   
17          56         8  0.105128  0.120099   0.0614   1.0513              8   
18          16         8  0.119093  0.129123   0.0743   1.1909              8   
19          24         8  0.119093  0.129123   0.0743   1.1909              8   
20          32         8  0.191727  0.218663   0.1094   1.9173              8   
21          40         8  0.191727  0.218663   0.1094   1.9173              8   
22           0         8  0.212500  0.234187   0.1325   2.1250              8   
23           8         8  0.212500  0.234187   0.1325   2.1250              8   
24          64         8  0.312500  0.679154   0.1744   3.1250              8   
25          72         8  0.312500  0.679154   0.1744   3.1250              8   
26          48         8  0.999148  1.301613   0.5569   9.9915              8   
27          56         8  0.999148  1.301613   0.5569   9.9915              8   
28          48         8  1.184014  1.463341   0.6756  11.8401              8   
29          56         8  1.184014  1.463341   0.6756  11.8401              8   
30          32         8  2.164062  2.614865   1.2508  21.6406              8   
31          40         8  2.164062  2.614865   1.2508  21.6406              8   

    ForecastL1  ForecastL2  ForecastMAPE  ForecastMASE  TestCount    TestL1  \
0     0.032658    0.049837        0.0177        0.3266          8  0.032658   
1     0.032658    0.049837        0.0177        0.3266          8  0.032658   
2     0.033130    0.046636        0.0183        0.3313          8  0.033130   
3     0.033130    0.046636        0.0183        0.3313          8  0.033130   
4     0.049051    0.059106        0.0288        0.4905          8  0.049051   
5     0.049051    0.059106        0.0288        0.4905          8  0.049051   
6     0.056250    0.072349        0.0313        0.5625          8  0.056250   
7     0.056250    0.072349        0.0313        0.5625          8  0.056250   
8     0.073257    0.080617        0.0444        0.7326          8  0.073257   
9     0.073257    0.080617        0.0444        0.7326          8  0.073257   
10    0.087500    0.106066        0.0560        0.8750          8  0.087500   
11    0.087500    0.106066        0.0560        0.8750          8  0.087500   
12    0.087500    0.106066        0.0560        0.8750          8  0.087500   
13    0.087500    0.106066        0.0560        0.8750          8  0.087500   
14    0.087500    0.106066        0.0560        0.8750          8  0.087500   
15    0.087500    0.106066        0.0560        0.8750          8  0.087500   
16    0.105128    0.120099        0.0614        1.0513          8  0.105128   
17    0.105128    0.120099        0.0614        1.0513          8  0.105128   
18    0.119093    0.129123        0.0743        1.1909          8  0.119093   
19    0.119093    0.129123        0.0743        1.1909          8  0.119093   
20    0.191727    0.218663        0.1094        1.9173          8  0.191727   
21    0.191727    0.218663        0.1094        1.9173          8  0.191727   
22    0.212500    0.234187        0.1325        2.1250          8  0.212500   
23    0.212500    0.234187        0.1325        2.1250          8  0.212500   
24    0.312500    0.679154        0.1744        3.1250          8  0.312500   
25    0.312500    0.679154        0.1744        3.1250          8  0.312500   
26    0.999148    1.301613        0.5569        9.9915          8  0.999148   
27    0.999148    1.301613        0.5569        9.9915          8  0.999148   
28    1.184014    1.463341        0.6756       11.8401          8  1.184014   
29    1.184014    1.463341        0.6756       11.8401          8  1.184014   
30    2.164062    2.614865        1.2508       21.6406          8  2.164062   
31    2.164062    2.614865        1.2508       21.6406          8  2.164062   

      TestL2  TestMAPE  TestMASE  
0   0.049837    0.0177    0.3266  
1   0.049837    0.0177    0.3266  
2   0.046636    0.0183    0.3313  
3   0.046636    0.0183    0.3313  
4   0.059106    0.0288    0.4905  
5   0.059106    0.0288    0.4905  
6   0.072349    0.0313    0.5625  
7   0.072349    0.0313    0.5625  
8   0.080617    0.0444    0.7326  
9   0.080617    0.0444    0.7326  
10  0.106066    0.0560    0.8750  
11  0.106066    0.0560    0.8750  
12  0.106066    0.0560    0.8750  
13  0.106066    0.0560    0.8750  
14  0.106066    0.0560    0.8750  
15  0.106066    0.0560    0.8750  
16  0.120099    0.0614    1.0513  
17  0.120099    0.0614    1.0513  
18  0.129123    0.0743    1.1909  
19  0.129123    0.0743    1.1909  
20  0.218663    0.1094    1.9173  
21  0.218663    0.1094    1.9173  
22  0.234187    0.1325    2.1250  
23  0.234187    0.1325    2.1250  
24  0.679154    0.1744    3.1250  
25  0.679154    0.1744    3.1250  
26  1.301613    0.5569    9.9915  
27  1.301613    0.5569    9.9915  
28  1.463341    0.6756   11.8401  
29  1.463341    0.6756   11.8401  
30  2.614865    1.2508   21.6406  
31  2.614865    1.2508   21.6406  
  Transformation                                              Model  \
0     Diff_D0102  Diff_D0102_LinearTrend_residue_zeroCycle_resid...   
1     Diff_D0102  Diff_D0102_LinearTrend_residue_bestCycle_byL2_...   
2     Diff_D0102  Diff_D0102_PolyTrend_residue_zeroCycle_residue...   
3     Diff_D0102  Diff_D0102_PolyTrend_residue_bestCycle_byL2_re...   
4  RelDiff_D0102  RelDiff_D0102_PolyTrend_residue_zeroCycle_resi...   

   Complexity  FitCount     FitL1     FitL2  FitMAPE  FitMASE  ForecastCount  \
0          48         8  0.032658  0.049837   0.0177   0.3266              8   
1          56         8  0.032658  0.049837   0.0177   0.3266              8   
2          48         8  0.033130  0.046636   0.0183   0.3313              8   
3          56         8  0.033130  0.046636   0.0183   0.3313              8   
4          48         8  0.049051  0.059106   0.0288   0.4905              8   

   ForecastL1  ForecastL2  ForecastMAPE  ForecastMASE  TestCount    TestL1  \
0    0.032658    0.049837        0.0177        0.3266          8  0.032658   
1    0.032658    0.049837        0.0177        0.3266          8  0.032658   
2    0.033130    0.046636        0.0183        0.3313          8  0.033130   
3    0.033130    0.046636        0.0183        0.3313          8  0.033130   
4    0.049051    0.059106        0.0288        0.4905          8  0.049051   

     TestL2  TestMAPE  TestMASE  
0  0.049837    0.0177    0.3266  
1  0.049837    0.0177    0.3266  
2  0.046636    0.0183    0.3313  
3  0.046636    0.0183    0.3313  
4  0.059106    0.0288    0.4905  
6 0    1.2
1    1.3
2    1.3
3    1.3
4    1.3
5    1.2
Name: D0102, dtype: float64
6 0    1.190247
1    1.075549
2    0.955907
3    0.831319
4    0.701786
5    0.567308
Name: D0102_Forecast, dtype: float64
FORECAST_DETAIL_ACTUAL D0102 D0102 [ 1.2  1.3  1.3  1.3  1.3  1.2]
FORECAST_DETAIL_PREDICTED D0102 D0102 [ 1.19024725  1.07554945  0.95590659  0.83131868  0.70178571  0.56730769]
BENCHMARK_PERF_DETAIL D0102 D0102 8 6 5.169441223144531 LinearTrend + NoCycle + NoAR 6 0.2989 0.3759 9.4912 0.379647435897 0.43716806532 -85.0021621561
BENCHMARK_SUCCESS 'M4_COMP_DEMOGRAPHICS_D0102_6'
